lib/fetch/formdata.js

```diff
@@ -0,0 +1,131 @@
+'use strict'
+
+const { Blob } = require('buffer')
+const { kState } = require('./symbols')
+const { File } = require('./file')
+
+class FormData {
+  constructor () {
+    this[kState] = []
+  }
+
+  append (name, value, filename) {
+    // 1. Let value be value if given; otherwise blobValue.
+    // 2. Let entry be the result of creating an entry with
+    // name, value, and filename if given.
+    const entry = makeEntry(name, value, filename)
+
+    // 3. Append entry to this’s entry list.
+    this[kState].push(entry)
+  }
+
+  delete (name) {
+    // The delete(name) method steps are to remove all entries whose name
+    // is name from this’s entry list.
+    const next = []
+    for (const entry of this[kState]) {
+      if (entry.name !== name) {
+        next.push(entry)
+      }
+    }
+
+    this[kState] = next
+  }
+
+  get (name) {
+    // 1. If there is no entry whose name is name in this’s entry list,
+    // then return null.
+    const idx = this[kState].findIndex((entry) => entry.name === name)
+    if (idx === -1) {
+      return null
+    }
+
+    // 2. Return the value of the first entry whose name is name from
+    // this’s entry list.
+    return this[kState][idx].value
+  }
+
+  getAll (name) {
+    // 1. If there is no entry whose name is name in this’s entry list,
+    // then return the empty list.
+    // 2. Return the values of all entries whose name is name, in order,
+    // from this’s entry list.
+    return this[kState]
+      .filter((entry) => entry.name === name)
+      .map((entry) => entry.value)
+  }
+
+  has (name) {
+    // The has(name) method steps are to return true if there is an entry
+    // whose name is name in this’s entry list; otherwise false.
+    return this[kState].findIndex((entry) => entry.name === name) !== -1
+  }
+
+  set (name, value, filename) {
+    // The set(name, value) and set(name, blobValue, filename) method steps
+    // are:
+
+    // 1. Let value be value if given; otherwise blobValue.
+    // 2. Let entry be the result of creating an entry with name, value, and
+    // filename if given.
+    const entry = makeEntry(name, value, filename)
+
+    // 3. If there are entries in this’s entry list whose name is name, then
+    // replace the first such entry with entry and remove the others.
+    const idx = this[kState].findIndex((entry) => entry.name === name)
+    if (idx !== -1) {
+      this[kState][idx] = entry
+      this[kState] = this[kState].filter((entry) => entry.name !== name)
+    } else {
+      // 4. Otherwise, append entry to this’s entry list.
+      this[kState].push(entry)
+    }
+  }
+
+  * [Symbol.iterator] () {
+    // The value pairs to iterate over are this’s entry list’s entries with
+    // the key being the name and the value being the value.
+    for (const { name, value } of this[kState]) {
+      yield [name, value]
+    }
+  }
+}
+
+function makeEntry (name, value, filename) {
```

Comment:
Do not forget to cast name and value (in case if it not Blob or File) to a string.

Reply:
Fixed